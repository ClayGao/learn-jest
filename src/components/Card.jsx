import { useCallback, useState } from 'react'

function Card({name, age, description}) {
  const [cardText, setCardText] = useState('')

  const onCardClick = useCallback(() => {
    setCardText(`已點擊`)
  },[])

  return (
    <div 
      onClick={onCardClick}
      data-testid="card-container" 
      className="round-sm border-2 h-full  flex-1"
    >
      <div>{name}</div>
      <div>{age}</div>
      <div>{description}</div>
      <div data-testid="card-text">{cardText}</div>
      <div>123</div>
    </div>
  );
}

export default Card;
