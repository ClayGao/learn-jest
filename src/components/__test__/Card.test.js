import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Card from '../Card'
import '@testing-library/jest-dom/extend-expect';

/**
 * fireEvent: 模擬 trigger 事件
 */


describe('<Card />', () => {
  // https://positivegrid-js.s3.amazonaws.com/contentful/sandbox/stickyBar.json?t=2021_08_19_09_55
  it('should render <Card /> correct', async () => {
    const { asFragment, getByTestId } = render(<Card />);

    const cardContainer = getByTestId('card-container')
    const cardContent = getByTestId('card-text')
    fireEvent.click(cardContainer)
    // textContent 取得 DOM 元素的內容
    expect(cardContent.textContent).toBe('已點擊')


    expect(asFragment()).toMatchSnapshot();
  });
});

