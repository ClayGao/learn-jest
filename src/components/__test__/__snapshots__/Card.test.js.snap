// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render <Card /> correct 1`] = `
<DocumentFragment>
  <div
    class="round-sm border-2 h-full  flex-1"
    data-testid="card-container"
  >
    <div />
    <div />
    <div />
    <div
      data-testid="card-text"
    >
      已點擊
    </div>
    <div>
      123
    </div>
  </div>
</DocumentFragment>
`;
